### PR TITLE
[azure] enable des in day2

### PR DESCRIPTION
**Pull request by jinyunma**
https://issues.redhat.com/browse/OCPQE-20769

Enable disk encryption set by using system managed key in day2

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

File: ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ref:
+  as: azure-enable-encryptionathost-day2
+  from: upi-installer
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.yaml:3]: Suggest to use a tag name of upi-installer image.

File: ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+	"path": "azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.metadata.json:2]: Maybe we can reuse the existing folder, two options:
1. ci-operator/step-registry/azure/provision/disk-encryption-set/day2/system-managed-key
2. ci-operator/step-registry/ipi/install/post/azure/disk-encryption-set/system-managed-key

WDYT?

File: ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.yaml (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.yaml]: updated

File: ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.metadata.json (hunk 2 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/enable-encryptionathost-day2/azure-enable-encryptionathost-day2-ref.metadata.json]: good suggestion, moved to ci-operator/step-registry/azure/provision/disk-encryption-set/day2/system-managed-key

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml [hunk 3]
```diff
@@ -1234,7 +1234,7 @@ tests:
       OCP_ARCH: arm64
     test:
     - chain: openshift-e2e-test-qe
-    workflow: cucushift-installer-rehearse-azure-ipi-post-featureset
+    workflow: cucushift-installer-rehearse-azure-ipi-post-featureset-des
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml:1237]: So for now, the day2 enable des is only available in TP cluster?

Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml]: It's unrelated with TP, move out from TP job, add new job `azure-ipi-des-day2`, workflow is `cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2`

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/disk-encryption-set/day2/provision/cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2-provision-chain.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2-provision
+  steps:
+  - chain: ipi-conf-azure
+  - chain: ipi-install
+  - ref: ipi-install-post-featureset
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/disk-encryption-set/day2/provision/cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2-provision-chain.yaml:6]: I guess we do not need this step.

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/disk-encryption-set/day2/provision/cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2-provision-chain.yaml]: right, removed.

**Comment by jinyunma** [comment]
new step `azure-provision-disk-encryption-set-day2-system-managed-key` has been verified via [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/50674/rehearse-50674-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure/1777212913874898944), and passed.

/pj-rehearse ack

cc @xialiang to review

**Pull request merged**